src/public-lib/lib/persistentApiClientCache.ts

```diff
@@ -0,0 +1,130 @@
+import { set, get, keys, del } from 'idb-keyval';
+import request from 'superagent';
+import internalClient from './../../shared/api/cbioportalInternalClientInstance';
+
+export type EndpointUpdateInfo = {
+    table: string,
+    endpoint: string,
+    timestamp?: Date,
+}
+type TimestampedReponse = { timestamp: Date } & request.Response;
+type Log = (message: string) => void;
+
+export const PERSISTENT_CACHE_PREFIX = "persistent-cache-"; // prefix to namespace keys from anything else in store
+var endpointUpdateTimes: Map<string, EndpointUpdateInfo> | Promise<void>;
+
+function createCacheCheckingVersionOfGet(name: string, api: any, log?: Log): void {
+    const oldMethod = api[name];
+    const method = async function(parameters: any): Promise<request.Response> {
+        // Wait for timestamps to finish loading to avoid race condition
+        // where the persistent cache would sometimes be ignored
+        if (endpointUpdateTimes instanceof Promise) {
+            await endpointUpdateTimes;
+        }
+
+        // for every GET in the api, there is a corresponding
+        // GETURL function that returns the url as a string
+        const urlMethod = constructUrlMethodName(name, log);
```

Comment:
i wanted to float the idea of just hashing the parameters (as we do in existing memory cache).  You could potentially add the timestamp to that has (would this obviate the need to store the timestamps elsewhere?).  The other possible benefit is that it would work for both post and get requests.

Reply:
Ok, I'm interested. I'll give it a shot. I don't like hashes because it makes understanding exactly what is in the cache difficult, but the added utility of being able to cache posts is probably worth it.